typescript/infra/src/agents/aws/validator.ts

```diff
@@ -0,0 +1,124 @@
+import { BaseValidator, types, utils } from '@abacus-network/utils';
+
+import { S3Receipt, S3Wrapper } from './s3';
+
+enum CheckpointStatus {
+  EXTRA = '➕',
+  MISSING = '❓',
+  INVALID = '❌',
+  VALID = '✅',
+}
+
+interface CheckpointMetric {
+  status: CheckpointStatus;
+  delta?: number;
+  violation?: string;
+}
+
+// TODO: merge with types.Checkpoint
+interface S3Checkpoint {
```

Comment:
Is it hard enough to not do in this PR?

Reply:
we need some interface which describes the shape of the data in S3
I didn't want to change all of the existing usage of the checkpoint type so doing a simple transformation when parsing for now